src/aof.c

```diff
@@ -2321,6 +2324,10 @@ int rewriteAppendOnlyFileBackground(void) {
 }
 
 void bgrewriteaofCommand(client *c) {
+    if (server.aof_state == AOF_OFF) {
+        addReplyError(c, "Can't execute an AOF background rewriting when AOF is disable.");
+        return;
```

Comment:
@itamarhaber @yossigo can you think of any use case that we're breaking now?

Reply:
Someone will use `BGREWRITEAOF` for snapshot backup (it enables preamble but it doesn't use `bgsave` command), I'm sure someone will.

Reply:
you mean that people can choose to use BGREWRITEAOF to create a snapshot in AOF (RESP) format (by disabling preamble rdb)?
but what's the use case for it? pipe that to redis-cli and load into another server?

Reply:
He set preamble to yes, then use `BGREWRITEAOF` to create a RDB format AOF.  At this time, the products brought by `bgsave` and `BGREWRITEAOF` are the same ？

Reply:
so why do that? why not use BGSAVE?
also, even if someone did that and we're breaking him, he can easily solve it by switching to BGSAVE

Reply:
I know my insistence may be useless, but I want to leave something in this PR so that if users come looking for a reason in the future, they can at least see that I fought for them.